index.ts

```diff
@@ -732,6 +786,190 @@ export const ReleaseCollection = {
+        const issue = repo.issues.one({ number: event.issue.number });
+        await issue.closed.$emit();
+      }
+
+      if (event.action === "created" && event.release) {
+        const release = repo.release.one({ id: event.release.id });
+        await repo.releasePublished.$emit({ release });
+      }
+
+      if (event.action === "opened" && event.pull_request) {
+        const pullRequest = repo.pull_request.one({ number: event.pull_request.number });
+        await repo.pullRequestOpened.$emit({ pullRequest });
+      }
+
+      if (event.action === "closed" && event.pull_request.closed) {
+        const pullRequest = repo.pull_request.one({ number: event.pull_request.number });
+        await pullRequest.closed.$emit();
+      }
+
+      if(event.action === "created" && event.comment){
+          const comment = repo.issues.one({ number: event.issue.number }).comments.one({ id: event.comment.id });
+          await repo.issues.one({ number: event.issue.number }).commentCreated.$emit({ comment });
+          await repo.commentCreated.$emit({ comment });
+      }
+      return JSON.stringify({ status: 200 });
+    }
+    default:
+      console.log("Unknown Endpoint:", path);
+  }
+}
+async function register(owner: string, repo: string, event: string) {
+  const webhookURL = (await nodes.endpoint.$get()) + "/webhooks";
+  try {
+    // Check if the repository already has a webhook
+    const repository = state.repos[`${owner}/${repo}`];
+    const { data: hooks } = await client().repos.listWebhooks({
+      owner,
+      repo,
+    });
+    const matchingHook = hooks.find((hook) => hook.id === repository.id);
+    // If the repository already has a webhook, update it
+    if (matchingHook) {
+      if (repository.events.includes(event)) {
+        console.log("Webhook already exists with the same event.");
+        return;
+      } else {
+        const updatedEvents = [...repository.events, event];
+        await client().repos.updateWebhook({
+          owner,
+          repo,
+          hook_id: repository.id,
+          config: {
+            content_type: "json",
+            url: webhookURL,
+          },
+          events: updatedEvents,
+        });
+        // Update the events array in the repository object
+        repository.events = updatedEvents;
+        console.log("Webhook updated with new event.");
+        return;
+      }
+    }
+    // Create a new webhook
+    const {
+      data: { id: webhookId },
+    } = await client().repos.createWebhook({
+      owner,
+      repo,
+      events: [event],
+      config: {
+        content_type: "json",
+        url: webhookURL,
+      },
+    });
+
+    // Add the repository to the state with the webhook data
+    state.repos[`${owner}/${repo}`] = {
+      id: webhookId,
+      url: webhookURL,
+      events: [event],
+    };
+
+    console.log("New webhook created.");
+  } catch (error) {
+    throw new Error(error);
+  }
```

Comment:
Let's add some context in the error message and add the original as `{ cause: error }`